### PR TITLE
Pimpl for Transaction Response

### PR DESCRIPTION
**Pull request by Akvinikym**
Signed-off-by: Akvinikym <anarant12@gmail.com>

### Description of the Change

Pimpl idiom is introduced for Transaction Response class.

### Benefits

Faster build.

### Possible Drawbacks 

None.

### CODE REVIEW

File: shared_model/backend/protobuf/transaction_responses/proto_tx_response.hpp [hunk 1]
```diff
@@ -9,81 +9,51 @@
 #include <limits>
```
Comment (lebdron) [inline shared_model/backend/protobuf/transaction_responses/proto_tx_response.hpp]: Is it possible to define `max_priority` in cpp file, so that `limits` is not included in hpp file?

File: shared_model/backend/protobuf/transaction_responses/proto_tx_response.hpp [hunk 2]
```diff
@@ -9,81 +9,51 @@
...
 
 #include "backend/protobuf/transaction_responses/proto_concrete_tx_response.hpp"
```
Comment (lebdron) [inline shared_model/backend/protobuf/transaction_responses/proto_tx_response.hpp]: Should be included in cpp file instead.

File: shared_model/backend/protobuf/transaction_responses/impl/proto_tx_response.cpp [hunk 3]
```diff
@@ -6,83 +6,100 @@
...
-    template TransactionResponse::TransactionResponse(
-        TransactionResponse::TransportType &&);
+    struct TransactionResponse::Impl {
+      explicit Impl(TransportType &&ref) : proto_{std::move(ref)} {}
+      explicit Impl(const TransportType &ref) : proto_{ref} {}
+
+      detail::ReferenceHolder<TransportType> proto_;
+
+      const ProtoResponseVariantType variant_{[this] {
+        auto &&ar = *proto_;
```
Comment (igor-egorov) [inline shared_model/backend/protobuf/transaction_responses/impl/proto_tx_response.cpp]: && and ar?

File: shared_model/backend/protobuf/transaction_responses/proto_tx_response.hpp [hunk 4]
```diff
@@ -9,81 +9,51 @@
...
-      template <typename TxResponse>
-      explicit TransactionResponse(TxResponse &&ref);
+      using TransportType = iroha::protocol::ToriiResponse;
 
       TransactionResponse(const TransactionResponse &r);
-
       TransactionResponse(TransactionResponse &&r) noexcept;
 
+      explicit TransactionResponse(const TransportType &ref);
+      explicit TransactionResponse(TransportType &&ref);
```
Comment (igor-egorov) [inline shared_model/backend/protobuf/transaction_responses/proto_tx_response.hpp]: Why already existing constructors (above) are not marked with explicit?

File: test/module/shared_model/builders/protobuf/transaction_responses/proto_transaction_response_builder.cpp [hunk 5]
```diff
@@ -76,8 +65,7 @@ TYPED_TEST(ProtoTransactionStatusBuilderTest, TestStatusType) {
                       .build();
 
   ASSERT_NO_THROW(boost::apply_visitor(
-      framework::SpecifiedVisitor<StatusType>(),
-      response.get()));
+      framework::SpecifiedVisitor<StatusType>(), response.get()));
```
Comment (igor-egorov) [inline test/module/shared_model/builders/protobuf/transaction_responses/proto_transaction_response_builder.cpp]: Since you have been touching this code. Please rework with boost::get

File: shared_model/backend/protobuf/transaction_responses/proto_tx_response.hpp (hunk 4 above)
Comment (lebdron) [reply shared_model/backend/protobuf/transaction_responses/proto_tx_response.hpp]: https://stackoverflow.com/a/6759059

TL;DR - copy and move constructors are expected to be nonexplicit
